index.js

```diff
@@ -480,7 +481,15 @@ class MagicOccupancy {
     }
 
     _setOccupancyState (newVal) {
+        // Capture previous state and set the new state
+        this.previousModeState = this.modeState;
         this.modeState = newVal;
+        this.log.debug(`Previous state was ${this.previousModeState} and is now ${this.modeState}`)
+
+        // Log when the state changes to Occupied or Unoccupied
+        if (this.modeState == 'Occupied' && this.previousModeState == 'Unoccupied' || this.modeState == 'Unoccupied') {
+            this.occupancyLogging && this.log(`Setting state to ${this.modeState}`)
+        }
```

Comment:
I think this can be simplified a little bit - additionally, I think we should move this line "setting state" to come before we actually set the state (so put it at the beginning of the function). Those changes together would look like this:


```
        if (this.occupancyLogging && this.modeState != newVal) {
             this.log(`Setting state to ${newVal}`)
         }
```

Reply:
I did try with similar logic to begin with, but as my motion sensor triggered and cleared the state changes between UnoccupiedDelay / Occupied would be logged continuously as the delay timer started and stopped which still felt a bit too noisy.

```
[1/11/2023, 5:13:44 PM] [Office MagicOccupancy] Setting state to UnoccupiedDelay
[1/11/2023, 5:13:45 PM] [Office MagicOccupancy] Setting state to Occupied
[1/11/2023, 5:14:13 PM] [Office MagicOccupancy] Setting state to UnoccupiedDelay
[1/11/2023, 5:14:30 PM] [Office MagicOccupancy] Setting state to Occupied
[1/11/2023, 5:15:00 PM] [Office MagicOccupancy] Setting state to UnoccupiedDelay
```

Adding the logic to only log the state changes when the following conditions are met cleaned up the noise and would only log when the occupancy sensor state truly changed in Homekit:
- Switching from unoccupied to occupied
- Switching to unoccupied

Would this be sufficient? It would clean up the if statement and "this.log" line while still retaining the logic that keeps the logging from becoming noisy.
```
    _setOccupancyState (newVal) {
        // Capture the previous state and determine if the state change should be logged
        this.previousModeState = this.modeState;
        this.occupancyLoggingCondition = (newVal == 'Occupied' && this.previousModeState == 'Unoccupied' || newVal == 'Unoccupied')
        if (this.occupancyLogging && this.occupancyLoggingCondition) {
            this.log(`Setting state to ${newVal}`)
        }

        // Set the new state
        this.modeState = newVal;
        this.log.debug(`Previous state was ${this.previousModeState} and is now ${this.modeState}`)
```

Reply:
I think that makes sense, to still only log those _big_ status changes but still making the few little cleanup suggestions (like saying "setting state" before actually setting the state instead of after)

Reply:
Thanks! Those changes have been made in 6872963.